src/plugins/features.go

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package plugins
+
+import (
+	"github.com/google/uuid"
+	agent_config "github.com/nginx/agent/sdk/v2/agent/config"
+	"github.com/nginx/agent/sdk/v2/client"
+	sdkGRPC "github.com/nginx/agent/sdk/v2/grpc"
+	"github.com/nginx/agent/v2/src/core"
+	"github.com/nginx/agent/v2/src/core/config"
+	log "github.com/sirupsen/logrus"
+)
+
+type Features struct {
+	commander client.Commander
+	pipeline  core.MessagePipeInterface
+	conf      *config.Config
+	env       core.Environment
+	binary    core.NginxBinary
+	version   string
+}
+
+func NewFeatures(commander client.Commander, conf *config.Config, env core.Environment, binary core.NginxBinary, version string) *Features {
+	return &Features{
+		commander: commander,
+		conf:      conf,
+		env:       env,
+		binary:    binary,
+		version:   version,
+	}
+}
+
+func (f *Features) Init(pipeline core.MessagePipeInterface) {
+	log.Info("Features initializing")
+	f.pipeline = pipeline
+}
+
+func (f *Features) Close() {
+	log.Info("Features is wrapping up")
+}
+
+func (f *Features) Info() *core.Info {
+	return core.NewInfo("Features Plugin", "v0.0.1")
+}
+
+func (f *Features) Subscriptions() []string {
+	return []string{
+		core.EnableFeature,
+	}
+}
+
+func (f *Features) Process(msg *core.Message) {
+	log.Infof("--------> Process function in the features.go, %s %v", msg.Topic(), msg.Data())
+	switch data := msg.Data().(type) {
+	case string:
+		switch msg.Topic() {
+		case core.EnableFeature:
+			switch data {
```

Comment:
this could be refactored into a map

e.g. ... ```featureMap := map[string]func(){
	agent_config.FeatureMetrics: func() {
...```

then called like: 
```
if initFeature, ok := featureMap[data.(string)]; ok {
	initFeature()
}
```